Mathlib/MeasureTheory/Integral/IntegrableOn.lean

```diff
@@ -158,41 +167,48 @@ theorem IntegrableOn.left_of_union (h : IntegrableOn f (s ∪ t) μ) : Integrabl
 theorem IntegrableOn.right_of_union (h : IntegrableOn f (s ∪ t) μ) : IntegrableOn f t μ :=
   h.mono_set subset_union_right
 
-theorem IntegrableOn.union (hs : IntegrableOn f s μ) (ht : IntegrableOn f t μ) :
+theorem IntegrableOn.union [PseudoMetrizableSpace ε]
+    (hs : IntegrableOn f s μ) (ht : IntegrableOn f t μ) :
     IntegrableOn f (s ∪ t) μ :=
   (hs.add_measure ht).mono_measure <| Measure.restrict_union_le _ _
 
 @[simp]
-theorem integrableOn_union : IntegrableOn f (s ∪ t) μ ↔ IntegrableOn f s μ ∧ IntegrableOn f t μ :=
+theorem integrableOn_union [PseudoMetrizableSpace ε] :
+    IntegrableOn f (s ∪ t) μ ↔ IntegrableOn f s μ ∧ IntegrableOn f t μ :=
   ⟨fun h => ⟨h.left_of_union, h.right_of_union⟩, fun h => h.1.union h.2⟩
 
 @[simp]
-theorem integrableOn_singleton_iff {x : α} [MeasurableSingletonClass α] :
+theorem integrableOn_singleton_iff {f : α → ε'} {x : α}
+    [MeasurableSingletonClass α] (hfx : ‖f x‖ₑ ≠ ⊤) :
```

Comment:
```suggestion
    [MeasurableSingletonClass α] (hfx : ‖f x‖ₑ ≠ ⊤ := by finiteness) :
```
maybe?

Reply:
Good idea! I also added a version `integrableOn_singleton`, similarly to `integrableOn_const`.